include/OpenSpaceToolkit/Astrodynamics/EventCondition/COECondition.hpp

```diff
@@ -43,11 +43,19 @@ using ostk::astro::trajectory::State;
 class COECondition
 {
    public:
+    /// @brief                  Semi-Major Axis based constructor
+    ///
+    /// @param                  [in] aCriterion An enum indicating the criterion used to determine the Event Condition
+    /// @param                  [in] aFrameSPtr A frame in which the Element is to be computed
+    /// @param                  [in] aSemiMajorAxis A Semi-Major Axis
+    /// @param                  [in] aGravitationalParameter A gravitational parameter
+
     static RealCondition SemiMajorAxis(
```

Comment:
nit: I know it was a lot of work to propagate these, but do you think it's worth it?

To me, it seems like fine UX to leave all of these specific COE condition generators as absolute valued, and if someone wants a relative condition, they can make their own. 

This is really a nitpick though. To me, it seems a little cleaner if factory functions do exactly 1 thing rather than trying to make them fully featured. Mostly a personal preference thing here though, so curious if others chime in.

Reply:
Yeah that's fair. I think it's a balance between readibility and maintainability.
Adding N RelativeSemiMajorAxis etc. is cumbersome and just more Tests, Bindings, etc. etc.
Therefore more to maintain. But it might be more readable/ better UX

Let me play with the target Enum idea and see how far I get.